src/plugins/analysis/ikconfig/test/test_ikconfig.py

```diff
@@ -0,0 +1,164 @@
+        test_file = FileObject(file_path=os.path.join(TEST_DATA_DIR, 'synthetic/configs.ko'))
+        test_file.processed_analysis['file_type'] = dict(mime='application/octet-stream')
+        test_file.processed_analysis['software_components'] = dict(summary=['Linux Kernel'])
+
+        result = AnalysisPlugin.try_object_extract_ikconfig(test_file.binary)
+
+        assert(len(result) > 0)
+        assert(self.analysis_plugin.probably_kernel_config(result))
+
+    def test_internal_gz_decompressor_break_on_true_gz_file(self):
+        test_file = FileObject(file_path=os.path.join(TEST_DATA_DIR, 'configs/CONFIG.gz'))
+        decompressor = GZDecompressor()
+        assert(decompressor.decompress(test_file.binary) != b'')
+
+    def test_process_valid_plain_text_config(self):
+        test_file = FileObject(file_path=os.path.join(TEST_DATA_DIR, 'configs/CONFIG'))
+        test_file.processed_analysis['file_type'] = dict(mime='text/plain')
+
+        self.analysis_plugin.process_object(test_file)
+
+        assert(test_file.processed_analysis[self.PLUGIN_NAME]['is_kernel_config'])
+        assert(test_file.processed_analysis[self.PLUGIN_NAME]['kernel_config'] == test_file.binary.decode('utf-8'))
+
+    def test_process_invalid_plain_text_config(self):
+        test_file = FileObject(file_path=os.path.join(TEST_DATA_DIR, 'random_invalid/c.image'))
+        test_file.processed_analysis['file_type'] = dict(mime='text/plain')
+
+        self.analysis_plugin.process_object(test_file)
+
+        assert('is_kernel_config' not in test_file.processed_analysis[self.PLUGIN_NAME])
+        assert('kernel_config' not in test_file.processed_analysis[self.PLUGIN_NAME])
+
+    def test_process_configs_ko_success(self):
+        test_file = FileObject(file_path=os.path.join(TEST_DATA_DIR, 'synthetic/configs.ko'))
+        test_file.processed_analysis['file_type'] = dict(mime='text/plain')
+
+        self.analysis_plugin.process_object(test_file)
+
+        assert(test_file.processed_analysis[self.PLUGIN_NAME]['is_kernel_config'])
+        assert(len(test_file.processed_analysis[self.PLUGIN_NAME]['kernel_config']) > 0)
+
+    def test_process_configs_ko_failure(self):
+        test_file = FileObject(file_path=os.path.join(TEST_DATA_DIR, 'synthetic/ko_failure/configs.ko'))
+        test_file.processed_analysis['file_type'] = dict(mime='text/plain')
+
+        self.analysis_plugin.process_object(test_file)
+
+        assert('is_kernel_config' not in test_file.processed_analysis[self.PLUGIN_NAME])
+        assert('kernel_config' not in test_file.processed_analysis[self.PLUGIN_NAME])
+
+    def test_process_objects_kernel_image(self):
+        synth_pattern = 'synthetic/*.image'
+        bad_pattern = 'random_invalid/*image'
+
+        valid_images = glob.glob(os.path.join(TEST_DATA_DIR, synth_pattern))
+        bad_images = glob.glob(os.path.join(TEST_DATA_DIR, bad_pattern))
+
+        for v in valid_images:
```

Comment:
pathlib + more readable variable name

```suggestion
        for image in TEST_DATA_DIR.glob(synth_pattern):
```